website_snippet_country_dropdown/views/snippets.xml

```diff
@@ -54,7 +54,7 @@
 <template id="country_dropdown" name="Country Dropdown">
     <t t-set="default_value" t-value="default_value or ''"/>
     <t t-set="countries" t-value="request.env['res.country'].search([])"/>
-    <t t-set="default_country" t-value="default_country or countries.search([('code', '=', default_value[:2])]) or website.company_id.country_id or countries[:1]"/>
+    <t t-set="default_country" t-value="default_country or countries.search([('code', '=', default_value[:2])]) or request.env.user.country_id or website.company_id.country_id or countries[:1]"/>
```

Comment:
As you can see, `default_value` defaults to an empty string. Odoo saves empty values as `False`, not as empty strings, so searching for empty strings will always return a empty recordset, unless your database is broken which is not this module's problem IMO.

You can test that by going to sales > localization > countries, creating a country with an empty code, and searching for "country code is equal to (and leave search box blank)".